pkg/mixin/provider/install.go

```diff
@@ -0,0 +1,84 @@
+package mixinprovider
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"path/filepath"
+	"runtime"
+
+	"github.com/deislabs/porter/pkg/mixin"
+	"github.com/pkg/errors"
+)
+
+func (p *FileSystem) Install(opts mixin.InstallOptions) (mixin.Metadata, error) {
+	mixinsDir, err := p.GetMixinsDir()
+	if err != nil {
+		return mixin.Metadata{}, err
+	}
+	mixinDir := filepath.Join(mixinsDir, opts.Name)
+
+	clientUrl := opts.GetParsedURL()
+	clientUrl.Path = path.Join(clientUrl.Path, opts.Version, fmt.Sprintf("%s-%s-%s%s", opts.Name, runtime.GOOS, runtime.GOARCH, mixin.FileExt))
+	clientPath := filepath.Join(mixinDir, opts.Name) + mixin.FileExt
+	err = p.downloadFile(clientUrl, clientPath)
+	if err != nil {
+		return mixin.Metadata{}, err
+	}
+
+	runtimeUrl := opts.GetParsedURL()
+	runtimeUrl.Path = path.Join(runtimeUrl.Path, opts.Version, fmt.Sprintf("%s-runtime-linux-amd64", opts.Name))
+	runtimePath := filepath.Join(mixinDir, opts.Name+"-runtime")
+	err = p.downloadFile(runtimeUrl, runtimePath)
+	if err != nil {
+		return mixin.Metadata{}, err
+	}
+
+	m := mixin.Metadata{
+		Name:       opts.Name,
+		Dir:        mixinDir,
+		ClientPath: clientPath,
+	}
+	return m, nil
+}
+
+func (p *FileSystem) downloadFile(url url.URL, destPath string) error {
+	if p.Debug {
+		fmt.Fprintf(p.Err, "Downloading %s to %s\n", url.String(), destPath)
+	}
+
+	resp, err := http.Get(url.String())
+	if err != nil {
+		return errors.Wrapf(err, "error downloading the mixin from %s", url.String())
+	}
+	if resp.StatusCode != 200 {
+		return errors.Errorf("bad status returned when downloading the mixin from %s (%d)", url.String(), resp.StatusCode)
+	}
+	defer resp.Body.Close()
+
+	// Ensure the parent directories exist
+	parentDir := filepath.Dir(destPath)
+	err = p.FileSystem.MkdirAll(parentDir, 0755)
+	if err != nil {
+		errors.Wrapf(err, "unable to create parent directory %s", parentDir)
+	}
+	defer p.FileSystem.RemoveAll(parentDir) // If we can't install the mixin, don't leave traces of it
```

Comment:
This area may still need attention... currently, I see:
```
 $ porter mixin install helm --url https://deislabs.blob.core.windows.net/porter/mixins/helm
Error: could not run mixin command /Users/vdice/.porter/mixins/helm/helm /Users/vdice/.porter/mixins/helm/helm version: fork/exec /Users/vdice/.porter/mixins/helm/helm: no such file or directory
```
Is it the `defer` statement?  Is there a way to only have `defer` run if err != nil?

Reply:
🤦‍♀️ zomg what was I thinking? That made perfect sense for some reason in the moment. 😂

Reply:
I miss mega `try/catch/finally` blocks...

Reply:
Fixed